src/loader/struct_util.rs

```diff
@@ -24,6 +31,30 @@ pub type Result<T> = std::result::Result<T, Error>;
 /// * `f` - The input to read from.  Often this is a file.
 /// * `out` - The struct to fill with data read from `f`.
 ///
+/// # Examples
```

Comment:
Since this is not a public module, can we also drop the examples? They are a bit superfluous for functions that only have 2 lines of meaningful code.

Reply:
Same as above, I think it should be public, but in another crate.

Reply:
I don't really mind it being public in another crate, but I don't really see the point in the linux loader exporting a `struct_util` module that doesn't really have anything to do with loading kernels. How about for now we keep it private and create a PR in vmm-sys-util instead with this utility and its documentation?

Reply:
That's where I was aiming too, this would have been an intermediary step but I agree with skipping it and going straight to `vmm-sys-util`.
BTW, `cmdline` is in the same situation. There's no point for `linux-loader` to publicly export functions that compose strings from key-value pairs.